core/models.py

```diff
@@ -436,18 +467,52 @@ def __str__(self):
     def save(self, *args, **kwargs):
         if self.start_date and timezone.is_naive(self.start_date):
             self.start_date = timezone.make_aware(self.start_date)
-        if self.end_date and timezone.is_naive(self.end_date):
+        elif self.end_date and timezone.is_naive(self.end_date):
             self.end_date = timezone.make_aware(self.end_date)
 
+        if self.review_status == ReviewStatus.Approved:
+            self.is_public = True
+        elif self.review_status == ReviewStatus.Rejected:
+            self.is_public = False
+
+        if self.duplication_id is None:
+            self.duplication_id = self.name.lower().replace(
+                " ", ""
+            ) + self.end_date.strftime("-%Y")
+        else:
+            pass
+
         super().save(*args, **kwargs)
 
 
+class Vote(models.Model):
+    vote_type = models.BooleanField(default=True)
```

Comment:
what does "True" mean. clarify naming to indicate meaning. e.g. positive or `is_upvote`